specs/experimental/validator-system-v2/validator-manager.md

```diff
@@ -0,0 +1,173 @@
+  - [Validator Tree](#validator-tree)
+  - [Status of Validator](#status-of-validator)
+  - [Jail](#jail)
+- [Entry Point](#entry-point)
+  - [Slashing](#slashing)
+  - [Reward Distribution](#reward-distribution)
+- [Summary of Definitions](#summary-of-definitions)
+  - [Constants](#constants)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The Validator Manager manages the set of [validators][g-validator] and selects the next priority validator with the
+priority to submit the [output root][g-l2-output]. It is also the entry point for other contracts, such as the
+[L2 Output Oracle][output-oracle] and the [Colosseum][colosseum-contract], which distribute
+[output rewards][g-validator-reward] and slash the challenge losers. It makes successive calls to the
+[Asset Manager contract](./asset-manager.md) to apply changes to the validators' assets.
+
+## Validator Registration
+
+To become a validator, the first step is to register to the Validator Manager with at least `MIN_REGISTER_AMOUNT` of
+tokens (KRO). Upon registration, they initiate their [`Vault`](./asset-manager.md#composition-of-asset-manager) and set
+commission rate and maximum change rate of the commission rate. The commission rate is the percentage that the validator
+takes from the output reward, and the maximum change rate is the maximum range of the commission rate that the validator
+can change at one time. If `COMMISSION_RATE_MIN_CHANGE_SECONDS` has not elapsed since the last time the validator
+changed the commission rate, the validator cannot change it again. These information and safeguards can help delegators
+choose which validators to delegate to.
+
+The `Validator` struct that represents the information of the validator is defined as follows:
+
+```solidity
+struct Validator {
+    bool isInitiated;
+    uint8 noSubmissionCount;
+    uint8 commissionRate;
+    uint8 commissionMaxChangeRate;
+    uint128 commissionRateChangedAt;
+}
+```
+
+After registration, validators can be delegated tokens and KGH NFTs. When the sum of delegated tokens exceeds
+`MIN_ACTIVATE_AMOUNT`, validators can activate its status and become eligible to submit output. When calculating the sum
+of delegated tokens, tokens contained in KGH are excluded as they are not subject to slashing. This means that
+validators should only fill up to `MIN_ACTIVATE_AMOUNT` with real tokens, not virtual ones.
+
+## Validator Management
+
+Once initiated, validators are managed using a data structure called validator tree and categorized statuses to select
+priority validator and check output submission eligibility. Below are the implementation details for managing
+validators.
+
+### Validator Tree
+
+To select next priority validator based on the amount of tokens delegated to the validator, validators are managed using
+a [self-balancing binary search tree structure][self-balancing-binary-search-tree]. When a validator invokes activate,
```

Comment:
```suggestion
a [self-balancing binary search tree structure][self-balancing-binary-search-tree]. When a validator invokes `activateValidator` to be activated,
```

Reply:
I think `activateValidator` is not mentioned before, so let's make simple.
```suggestion
a [self-balancing binary search tree structure][self-balancing-binary-search-tree]. When a validator is activated,
```